From: "Mike Rapoport (IBM)" <rppt@kernel.org>
Subject: mips: drop ranges for definition of ARCH_FORCE_MAX_ORDER
Date: Wed, 22 Mar 2023 10:15:20 +0200

MIPS defines insane ranges for ARCH_FORCE_MAX_ORDER allowing MAX_ORDER up
to 63, which implies maximal contiguous allocation size of 2^63 pages.

Drop bogus definitions of ranges for ARCH_FORCE_MAX_ORDER and leave it a
simple integer with sensible defaults.

Users that *really* need to change the value of ARCH_FORCE_MAX_ORDER will
be able to do so but they won't be mislead by the bogus ranges.

Link: https://lkml.kernel.org/r/20230322081520.2516226-1-rppt@kernel.org
Signed-off-by: Mike Rapoport (IBM) <rppt@kernel.org>
Acked-by: Thomas Bogendoerfer <tsbogend@alpha.franken.de>
Acked-by: Kirill A. Shutemov <kirill.shutemov@linux.intel.com>
Reviewed-by: David Hildenbrand <david@redhat.com>
Cc: Huacai Chen <chenhuacai@kernel.org>
Cc: WANG Xuerui <kernel@xen0n.name>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/arch/mips/Kconfig~mips-drop-ranges-for-definition-of-arch_force_max_order
+++ a/arch/mips/Kconfig
@@ -2137,13 +2137,9 @@ endchoice
 
 config ARCH_FORCE_MAX_ORDER
 	int "Maximum zone order"
-	range 13 63 if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_64KB
 	default "13" if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_64KB
-	range 12 63 if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_32KB
 	default "12" if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_32KB
-	range 11 63 if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_16KB
 	default "11" if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_16KB
-	range 0 63
 	default "10"
 	help
 	  The kernel memory allocator divides physically contiguous memory
_
